### PR TITLE
P2 231 be harvest logs correction

### PR DESCRIPTION
**Pull request by BryanCaldeira**
Correction Logs

### CODE REVIEW

File: controllers/harvest-log.controllers.ts [hunk 1]
```diff
@@ -27,10 +27,48 @@ const populateQuery = [
...
+      },
+    ],
+    select: ["-correctionLogs", "+createdAt"],
+  },
 ];
 
 const create = async (req: Request, res: Response) => {
   try {
+    const parentId = req.params?.id;
+
```
Comment (BryanCaldeira) [inline controllers/harvest-log.controllers.ts:46]: If id is passed in the route it is considered a correction log

Comment (BryanCaldeira) [reply controllers/harvest-log.controllers.ts:46]: Else a normal log

**Review by whamester — APPROVED** [review]

**Review by ReinhardtBotha — APPROVED** [review]
BoratBoratVeryNiceGIF

File: routes/harvest-logs.routes.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-import paths from "../shared/paths";
 import harvestLogValidator from "../shared/validators";
+import paths from "../shared/paths";
 
 const harvestLogRouter = express.Router();
 
-// POST: /api/v1/harvest-logs
+// POST: /api/v1/harvest-logs:id
 harvestLogRouter.post(
   paths.harvestLog,
```
Comment (whamester) [inline routes/harvest-logs.routes.ts:11]: you have to add the :id to the path

Screenshot 2024-03-08 at 3 36 38 PM

Comment (whamester) [reply routes/harvest-logs.routes.ts:11]: `${paths.harvestLog}/:id`

**Review by whamester — CHANGES_REQUESTED** [review]
change routes

File: routes/harvest-logs.routes.ts (hunk 2 above)
Comment (BryanCaldeira) [reply routes/harvest-logs.routes.ts:11]: This might have been caused during merge... I has it there... Will make the change

**Review by ALFAROSO — APPROVED** [review]

**Review by whamester — APPROVED** [review]

**Pull request merged**